dokka-subprojects/analysis-kotlin-symbols/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/symbols/translators/DRIFactory.kt

```diff
@@ -93,6 +93,28 @@ internal fun KtAnalysisSession.getDRIFromValueParameter(symbol: KtValueParameter
     return funDRI.copy(target = PointingToCallableParameters(index))
 }
 
+/**
+ * @return [DRI] to receiver type
+ */
+internal fun KtAnalysisSession.getDRIFromReceiverParameter(receiverParameterSymbol: KtReceiverParameterSymbol): DRI =
+    getDRIFromReceiverType(receiverParameterSymbol.type)
+
+private fun KtAnalysisSession.getDRIFromReceiverType(type: KtType): DRI {
+    return when(type) {
+        is KtNonErrorClassType -> getDRIFromNonErrorClassType(type)
+        is KtTypeParameterType -> getDRIFromTypeParameter(type.symbol)
+        is KtDefinitelyNotNullType -> getDRIFromReceiverType(type.original)
+        is KtTypeErrorType -> DRI(packageName = "", classNames = "$ERROR_CLASS_NAME $type")
+        is KtClassErrorType -> DRI(packageName = "", classNames = "$ERROR_CLASS_NAME $type")
+
+        is KtDynamicType -> throw IllegalStateException("Dynamic receiver is prohibited while creating DRI $type")
+        is KtCapturedType -> throw IllegalStateException("Unexpected non-denotable type while creating DRI $type")
```

Comment:
When I read these two lines, "unexpected" (from the second line) doesn't cause many questions, it likely means this type just hasn't been supported for whatever reason, but I can implement it if I need to fix a bug or something, so it's all good

"prohibited", however, immediately causes the question of "why". It reads as a strong requirement that it must not be implemented, and if I do implement it (in case there's a bug associated with it), something very bad will happen. and that if this error is even thrown in a user project, it's an indicator of a major problem.

is that the case? if it is, could you please add a code comment or change the exception message to explain why it's prohibited?